src/datatrove/pipeline/extractors/base.py

```diff
@@ -114,13 +127,28 @@ def process_document(self, text, extract_fn):
         self._ensure_process(extract_fn)
         try:
             self.parent_conn.send(text)
-            if self.parent_conn.poll(timeout=self.timeout):
-                result = self.parent_conn.recv()
-                if isinstance(result, Exception):
-                    raise result
-                return result
-            else:
-                raise TimeoutError("Document extraction timed out")
+
+            deadline = time.time() + self.timeout
```

Comment:
```suggestion
            deadline = time.monotonic() + self.timeout
```
time.monotonic is often used for timeouts, better resolution than time.time